department-of-veterans-affairs/caseflow
# RPO Queue - Assigned tab

Resolves [#{CASEFLOW-4602}](https://vajira.max.gov/browse/CASEFLOW-4602l)

### Description
  As a developer, to comply with the Education Pre-Docket requirements, I need understand the expected behavior of the RPO queue and its tabs and develop them. The RPO's assigned tab will represent all appeals that are sent to the RPO organization and have not been sent back to EMO or BVA intake and not marked in Progress. 
  For RPO there will be three tabs: Assigned, In Progress and Completed. This process applies for all the RPO offices: Buffalo RPO, Central Processing Office RPO and Muskogee RPO.

### Acceptance Criteria

- [ ] Determination of the behavior and triggers of the Assigned tab

- [ ] When appeals are assigned to the RPO organization, by the EMO organization, they show on the organization's Assigned tab. If an appeal is sent back to EMO or sent to BVA Intake, it no longer shows up on the RPO’s Assigned tab
When a user in the RPO organization selects the action, "Mark task In Progress", the appeal is removed from the Assigned tab. 

### Testing Plan
1.Intake a new 10182 appeal with a BVA Intake user (either BVADWISE or BVAISHAW)
2.When you reach the Add/Remove Issues screen, add an issue selecting "Education" as the benefit, selecting "Yes" for pre-docketing then submit the appeal
3.On the following success screen you should see a link to "edit issues" on the appeal, 
 go ahead and navigate to that and grab the uuid for the appeal which you'll find in the current URL
4. Type in make c into your rails console
5. Look up an uuid to save inside variable (appeal = Appeal.find_by_uuid("uuid")
6. Go to CASEFLOW test user page and Switch users
7. Go to the User Selector dropdown box and type in RPO click on EDURPOUSER
8. Create a EducationDocumentationTask
```ruby
rpo = EduRegionalProcessingOffice.first
# Where <id> is the Task ID from appeal.treee which correspond to the EducationDocumentSearchTask 
parent_task = Task.find(<id>)

rpo_task = EducationAssessDocumentationTask.create!( parent: parent_task, appeal: appeal, assigned_at: Time.zone.now, assigned_to: rpo )
```
9.Switch to Education RPO user then go to RPO Queue
10.Click on  Buffalo RPO from the dropdown
11.Make sure Buffalo RPO user case is on  Assigned  tab
12. Set task to in_progress and make sure it shows up in the in_progress tab
```ruby
rpo_task.update!(status: Constants.TASK_STATUSES.in_progress)

```
ruby
13.
When RPO task is updated to completed it will no longer show up on the Assigned or In Progress tab
```ruby
rpo_task.update!(status: Constants.TASK_STATUSES.completed)
parent_task.update!(status: Constants.TASK_STATUSES.assigned)

```

![Screenshot (55)](https://user-images.githubusercontent.com/18747251/165547411-08896b4c-ff4d-4349-9114-748c6347d841.png)
















### User Facing Changes

 
![Screenshot (55)](https://user-images.githubusercontent.com/18747251/165548932-3bc8c400-8bd7-4ca0-8c8e-80a479b345f6.png)

 

### Code Documentation Updates
